src/Tools/CoreFx.Tools/GenerateSlnFile.cs

```diff
@@ -0,0 +1,497 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text.RegularExpressions;
+using System.Xml;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class GenerateSlnFile : BuildTask
+    {
+        [Required]
+        public string PathToGenerate { get; set; }
+        [Required]
+        public string TemplateFile { get; set; }
+
+        private const string CONFIGURATION_PROPS = "Configurations.props";
```

Comment:
const's should just be Pascal-cased (`ConfigurationProps` or `ConfigurationPropsFileName`).